packages/xml-views-validation/src/validators/elements/non-stable-id.ts

```diff
@@ -0,0 +1,123 @@
+import { some, includes } from "lodash";
+import { XMLElement } from "@xml-tools/ast";
+import {
+  getUI5ClassByXMLElement,
+  ui5NodeToFQN,
+  resolveXMLNS,
+} from "@ui5-language-assistant/logic-utils";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { NonStableIDIssue } from "../../../api";
+import { NON_STABLE_ID, getMessage } from "../../utils/messages";
+import { isCustomClass } from "../../utils/custom-class";
+import { CORE_NS } from "../../utils/special-namespaces";
+
+export function validateNonStableId(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): NonStableIDIssue[] {
+  // Can't give an error if there is no position or name
+  if (xmlElement.name === null || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  if (isWhiteListed(xmlElement, model)) {
+    return [];
+  }
+
+  if (!isCustomClass(xmlElement)) {
```

Comment:
```typescript
if (!isCustomClass(xmlElement)) {
    return [];
  }
```

This reads like the check is only for custom Classes.
But logically the check is for either classes we identify (in model) OR customClasses.

So there is an assumption that `is[Possible]CustomClass` will return true for all classes in the model, which is probably correct now, but not necessarily in the future.

I'd rather have a more robust check:
```typescript
if (!isPossibleCustomClass(xmlElement) && !isKnownUI5Control/Element/Class) {
    return [];
  }
```